crates/typst/src/vm/opcodes_raw.rs

```diff
@@ -0,0 +1,507 @@
+// THIS IS PURPOSELY NOT ADDED IN A MODULE, BECAUSE IT IS USED SOMEWHERE ELSE.
+// THE GOAL IS TO KEEP ALL OPCODES IN ONE PLACE, SO THAT THEY CAN BE EASILY
+// REFERENCED.
+
+opcodes! {
+    // -----------------------------------------------------------------------------
+    // --------------------------------- OPERATORS ---------------------------------
+    // -----------------------------------------------------------------------------
+
+    /// No operation.
+    Nop: noop = 0x00,
+
+    /// Adds two values together.
+    Add: add -> Writable => {
+        /// The left-hand side of the addition.
+        lhs: Readable,
+        /// The right-hand side of the addition.
+        rhs: Readable,
+    } = 0x01,
+
+    /// Subtracts two values.
+    Sub: sub -> Writable => {
+        /// The left-hand side of the subtraction.
+        lhs: Readable,
+        /// The right-hand side of the subtraction.
+        rhs: Readable,
+    } = 0x02,
+
+    /// Multiplies two values.
+    Mul: mul -> Writable => {
+        /// The left-hand side of the multiplication.
+        lhs: Readable,
+        /// The right-hand side of the multiplication.
+        rhs: Readable,
+    } = 0x03,
+
+    /// Divides two values.
+    Div: div -> Writable => {
+        /// The left-hand side of the division.
+        lhs: Readable,
+        /// The right-hand side of the division.
+        rhs: Readable,
+    } = 0x04,
+
+    /// Negates a value.
+    Neg: neg -> Writable => {
+        /// The value to negate.
+        value: Readable,
+    } = 0x05,
+
+    /// Positivizes a value.
+    Pos: pos -> Writable => {
```

Comment:
is this Abs?

Reply:
It's weird, it was a expression in the old code, it just doesn't do anything unless your type is not numeric. Since I can't check for type information at compile-time (yet at least) I just made it an instruction.

Reply:
It's nice to be able to write unary `+1` when aligning with `-1`.